merkledag/ipld.md

```diff
@@ -72,6 +85,99 @@ O_5 = | "hello": "world"  |  whose hash value is QmR8Bzg59Y4FGWHeu9iTYhwhiP8PHCN
 
 This entire _merkle-path_ traversal is a unix-style path traversal over a _merkle-dag_ which uses _merkle-links_ with names.
 
+**[In case we use escaping in protobuf IPLD format]**
+
+In order to not restrict individual path component by disallowing some file names and still allow storing arbitrary data in IPLD objects, path components must be escaped when they are looked up in IPLD objects.
+
+To escape a path component in order to look it up in an IPLD object:
+
+- every `\` character in the path component must be replaced with `\\`
+- every `@` character in the path component must be replaced with `\@`
+
+This makes any key containing a `@` character unescaped in an IPLD object not accessible through a _filesystem merkle-path_. This is a reserved key that can be used to store auxiliary data without making it a link and visible in regular filesystems. This data can be made available in filesystems through extended attributes or opening and reading file contents.
+
+To unescape IPLD object keys that are not reserved and get the corresponding path component:
+
+- every `\@` sequence in the key must be replaced by `@`
+- every `\\` sequence in the key must be replaced by `\`
+
+
+### IPLD merkle-path (best solution)
+
+An _IPLD merkle-path_ is an extension of a _filesystem merkle-path_ which uses a special syntax to access link properties. **[In case we use escaping in protobuf IPLD format** Except that key escaping is not performed when looking up items in the IPLD objects. This allow accessing reserved keys using _IPLD merkle-paths_ that are not accessible in filesystems.**]**
+
+
+Path elements are suffixed by either `.link` to access the link properties or by `.object` to dereference the _merkle-link_. if no suffix is present, the _merkle-link_ is dereferenced (to be compatible with _filesystem merkle-paths_ in most cases)
```

Comment:
**Disclaimer:** I'm late to this party so if I missed something please excuse me

I do very much like the notion of this solution, but `.link` and `.object` seem quite verbose to me, why not use two separators like `#` and `.` instead, so the example paths would be transformed to the ones you can see below

- the path `/ipfs/QmCCC...000/stuff/foo#mode` yields `0755`
- the path `/ipfs/QmCCC...000/stuff/foo/other/cat#mode` does not exists because `other` does not have a `cat` object, only a `cat.link`
- the path `/ipfs/QmCCC...000/stuff/foo/other/cat.link.mode` yields `0644`
- the path `/ipfs/QmCCC...000/stuff/foo.other/cat#` yields object `QmCCC...222`

This would make the paths a little less verbose for me. I do see that conflicts with things like `cat.link` in the second object are still an issue, so it might make sense to use different separators, though `.` feels quite natural for the object selection to me.